### PR TITLE
Use RouterInterface / Remove clone router

### PR DESCRIPTION
**Pull request by hlohaus**
### 1. Why is this change necessary?

Use RouterInterface in Bundles for better decorating.
Remove clone router because less head.

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Shopware/Bundle/StoreFrontBundle/Service/Core/HrefLangService.php [hunk 1]
```diff
@@ -158,20 +158,18 @@ private function getLanguageShops(Shop $shop)
     /**
      * @param int $shopId
      *
-     * @return Router
+     * @return Context
+     *                 e
```
Comment (jkrzefski) [inline engine/Shopware/Bundle/StoreFrontBundle/Service/Core/HrefLangService.php]:
```suggestion
```

I think this line should not exist.

File: engine/Shopware/Bundle/StoreFrontBundle/Service/Core/HrefLangService.php [hunk 2]
```diff
@@ -158,20 +158,17 @@ private function getLanguageShops(Shop $shop)
     /**
      * @param int $shopId
      *
-     * @return Router
+     * @return Context
      */
-    private function getRouter($shopId)
+    private function getContext($shopId)
```
Comment (mitelg) [inline engine/Shopware/Bundle/StoreFrontBundle/Service/Core/HrefLangService.php:163]: why is this change necessary? 
imho it is not related to the goal of the PR to exchange the tyehints and should come in another PR

Comment (hlohaus) [reply engine/Shopware/Bundle/StoreFrontBundle/Service/Core/HrefLangService.php:163]: technically it does the same thing.
I just did not think it was so nice to clone the router.

Comment (hlohaus) [reply engine/Shopware/Bundle/StoreFrontBundle/Service/Core/HrefLangService.php:163]: In addition, it fixes that the config does not fit

Comment (mitelg) [reply engine/Shopware/Bundle/StoreFrontBundle/Service/Core/HrefLangService.php:163]: then please put that in a separate commit and change the description of your PR. This is nowhere mentioned

**Comment by hlohaus** [comment]
Okay.  There are now 2 commits and i updated the description
